lib/reek/smells/smell_repository.rb

```diff
@@ -25,6 +25,7 @@ def self.smell_types
       # @return [Array<Reek::Smells::SmellDetector>] All SmellDetectors that we want to filter for
       #         e.g. [Reek::Smells::Attribute].
       def self.eligible_smell_types(filter_by_smells = [])
+        filter_by_smells = Array(filter_by_smells)
```

Comment:
Hmmm. Hmmm. Well. It seems like you're trying to make up for Ruby's lack of static typing.
If you're doing this here, shouldn't we then consequently update ALL methods in Reek?
This change seems arbitrary - Why do we do this?